ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/Situation.java

```diff
@@ -0,0 +1,188 @@
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public class Situation {
+
+    private List<String> variantIds;
+
+    private String workingVariantId;
+
+    /**
+     * Network object
+     */
+    private Network network;
+
+    /**
+     * Crac object
+     */
+    private Crac crac;
+
+    /**
+     * Results of the systematic sensitivity analysis performed on the situation
+     */
+    private Map<String, SystematicSensitivityAnalysisResult> systematicSensitivityAnalysisResultMap;
+
+    /**
+     * constructor
+     */
+    public Situation(Network network, Crac crac) {
+        this.network = network;
+        this.crac = crac;
+        this.variantIds = new ArrayList<>();
+
+        ResultVariantManager resultVariantManager = crac.getExtension(ResultVariantManager.class);
+        if (resultVariantManager == null) {
+            resultVariantManager = new ResultVariantManager();
+            crac.addExtension(ResultVariantManager.class, resultVariantManager);
+        }
+
+        String situationVariantId = getUniqueSituationId();
+        network.getVariantManager().cloneVariant(network.getVariantManager().getWorkingVariantId(), situationVariantId);
+        crac.getExtension(ResultVariantManager.class).createVariant(situationVariantId);
+        variantIds.add(situationVariantId);
+        setWorkingVariant(situationVariantId);
+        init();
+
+        this.systematicSensitivityAnalysisResultMap = new HashMap<>();
+    }
+
+    /**
+     * Add in the Crac extension the initial RangeActions set-points
+     */
+    private void init() {
+        String preventiveState = getCrac().getPreventiveState().getId();
+        for (RangeAction rangeAction : getCrac().getRangeActions()) {
+            double valueInNetwork = rangeAction.getCurrentValue(network);
+            RangeActionResultExtension rangeActionResultMap = rangeAction.getExtension(RangeActionResultExtension.class);
+            RangeActionResult rangeActionResult = rangeActionResultMap.getVariant(workingVariantId);
+            rangeActionResult.setSetPoint(preventiveState, valueInNetwork);
+            if (rangeAction instanceof PstRange) {
+                ((PstRangeResult) rangeActionResult).setTap(preventiveState, ((PstRange) rangeAction).computeTapPosition(valueInNetwork));
+            }
+        }
+    }
+
+    /**
+     * get a unique network variant ID
+     */
+    private String getUniqueNetworkVariantId() {
+        String uniqueId;
+        do {
+            uniqueId = UUID.randomUUID().toString();
+        } while (network.getVariantManager().getVariantIds().contains(uniqueId));
+        return uniqueId;
+    }
+
+    private String getUniqueSituationId() {
+        String networkVariantId = getUniqueNetworkVariantId();
+        String cracVariantId = crac.getExtension(ResultVariantManager.class).getUniqueVariantId();
+        return String.valueOf(networkVariantId.concat(cracVariantId).hashCode());
```

Comment:
Don't use the hashCode of the concat. Two different strings may (very unlikely) have the same hashCode, so if you're very very unlucky you may end up with a situation variant id that already exists.

Reply:
ok